### DISCUSSION TITLE
Can we use GPU？

### DISCUSSION THREAD

**Discussion by SCUTE-ZZ** [ambros-gleixner/MIPcc23#13] [Q&A] [answered]
Is it possible to use a GPU? If possible, it is possible to use a deep learning model for heuristic selection b&b.

**Comment by ambros-gleixner** [comment, marked as answer]
_Hi SCUTE-ZZ,_

Let me answer the question first by emphasizing that the point of the competition is **not** to perform any pre-training on the general MIP structure of publicly available instances.  Your question sounds to me like you would want to do this in order to select heuristics depending on features of the MIP instances, independent of what you have observed from the solving process for previous instances from a series.

Whereas it would be generally okay to use this in order to craft your base MIP solver, it would probably lead to disqualification if your main contribution does not reuse information from previous solves as it processes the instance series.

This being said, we will not have reasonable GPU facilities on the evaluation machine, so also because of this your algorithm should limit itself to single-threaded CPU execution.

But possibly you could use different ML models?  Feel free to contact us also personally at mipcc-23@htw-berlin.de if you would like to discuss confidentially the viability of your approach and the needs you would have to execute it during evaluation.

Best regards,
_Ambros_
